src/Action.js

```diff
@@ -1,137 +1,131 @@
-import assert from './assert';
+import assert from './assert.js';
+import Field from './Field.js';
 import { contains, getMatchingValue, getMatchingValuesByAll, hasProperty } from './util.js';
-import Field from './Field';
 
-export default function Action(action) {
-	if (action instanceof Action) {
-		return action;
+export default class Action {
```

Comment:
Definitely hide whitespace changes for this diff. All I did here was the built-in "Convert to ES6 Class" in VS Code and that was it.

@marcsoiferman note that with this becoming classes, I believe that throws a wrench in your `Thing.prototype.extraMethod`-type extensions; sadly I think the only way around that is defining your own class that extends these ones, but that will likely run into some other problems. In the end, functions that take an instance of `Thing` as an argument might be the only sane option?